builder/dockerfile/evaluator.go

```diff
@@ -194,12 +236,14 @@ type dispatchState struct {
 	imageID    string
 	baseImage  builder.Image
 	stageName  string
+	shlex      *ShellLex
+	builder    *Builder
+	source     builder.Source
```

Comment:
These new fields are not really state, they are dependencies required by the dispatchers. I think they should remain in dispatchRequest, which would have a `state` field as well.